openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,212 @@
+                self.log.warning("Config file doesn't exist, skipping")
+                continue
+
+            repre = self._handle_original_repre(repre, profile)
+
+            for _, output_def in profile.get("outputs", {}).items():
+                new_repre = copy.deepcopy(repre)
+
+                new_staging_dir = get_transcode_temp_directory()
+                original_staging_dir = new_repre["stagingDir"]
+                new_repre["stagingDir"] = new_staging_dir
+                files_to_convert = new_repre["files"]
+                if not isinstance(files_to_convert, list):
+                    files_to_convert = [files_to_convert]
+
+                files_to_delete = copy.deepcopy(files_to_convert)
+
+                output_extension = output_def["output_extension"]
+                output_extension = output_extension.replace('.', '')
+                if output_extension:
+                    new_repre["name"] = output_extension
+                    new_repre["ext"] = output_extension
+
+                    files_to_convert = self._rename_output_files(
+                        files_to_convert, output_extension)
+
+                files_to_convert = [os.path.join(original_staging_dir, path)
+                                    for path in files_to_convert]
+
+                target_colorspace = output_def["output_colorspace"]
+                if not target_colorspace:
+                    raise RuntimeError("Target colorspace must be set")
+
+                convert_colorspace_for_input_paths(
+                    files_to_convert,
+                    new_staging_dir,
+                    config_path,
+                    source_color_space,
+                    target_colorspace,
+                    self.log
+                )
+
+                instance.context.data["cleanupFullPaths"].extend(
+                    files_to_delete)
+
+                custom_tags = output_def.get("custom_tags")
+                if custom_tags:
+                    if not new_repre.get("custom_tags"):
+                        new_repre["custom_tags"] = []
+                    new_repre["custom_tags"].extend(custom_tags)
+
+                # Add additional tags from output definition to representation
+                for tag in output_def["tags"]:
+                    if tag not in new_repre["tags"]:
+                        new_repre["tags"].append(tag)
+
+                instance.data["representations"].append(new_repre)
+
+    def _rename_output_files(self, files_to_convert, output_extension):
+        """Change extension of converted files."""
+        renamed_files = []
+        for file_name in files_to_convert:
+            file_name, _ = os.path.splitext(file_name)
+            new_file_name = '{}.{}'.format(file_name,
+                                           output_extension)
+            renamed_files.append(new_file_name)
+        files_to_convert = renamed_files
+        return files_to_convert
```

Comment:
```suggestion
        return renamed_files
```
We are not assigning in-place here so the original assignment to `files_to_convert` has no use here.